TMDS_3/include/CGAL/facets_in_complex_3_to_triangle_mesh.h

```diff
@@ -0,0 +1,245 @@
+    else
+    {
+      if(((cell_sdi == sd_index) == (s%2 == 1)) == normals_point_outside_of_the_subdomain)
+        std::swap(f[0], f[1]);
+    }
+
+    faces.push_back(f);
+    patches.push_back(spi);
+  }
+}
+
+template<class C3T3, class PointContainer, class FaceContainer, class SurfacePatchContainer>
+void facets_in_complex_3_to_triangle_soup(const C3T3& c3t3,
+                                          PointContainer& points,
+                                          FaceContainer& faces,
+                                          SurfacePatchContainer& patches)
+{
+  typedef typename C3T3::Subdomain_index              Subdomain_index;
+  Subdomain_index useless = Subdomain_index();
+  facets_in_complex_3_to_triangle_soup(c3t3, useless, points, faces, patches,
+                                       true/*point outward*/, true /*extract all facets*/);
+}
+
+template <typename Index2FaceMap, typename SurfacePatchRange>
+void set_face_patches(const Index2FaceMap&,
+                      const SurfacePatchRange&,
+                      const internal_np::Param_not_found&)
+{
+  return;
+}
+
+template <typename Index2FaceMap,
+          typename SurfacePatchRange,
+          typename FacePatchMap>
+void set_face_patches(const Index2FaceMap& i2f,
+                      const SurfacePatchRange& patches,
+                      const FacePatchMap& fpmap)
+{
+  for (auto index_and_face : i2f)
+  {
+    put(fpmap, index_and_face.second, patches[index_and_face.first]);
+  }
+}
+
+} // end namespace internal
+
+} // end namespace TMDS_3
+
+  /**
+   * @ingroup PkgTMDS3Functions
+   *
+   * @brief builds a `TriangleMesh` from the surface facets, with a consistent orientation at the interface of two subdomains.
+   *
+   * This function exports the surface as a `TriangleMesh` and appends it to `tm`, using `orient_polygon_soup()`.
+   *
+   * @tparam C3T3 a model of `MeshComplexWithFeatures_3InTriangulation_3`.
+   * @tparam TriangleMesh a model of `MutableFaceGraph` with an internal point property map. The point type must be compatible with the one used in `C3T3`.
+   * @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+   *
+   * @param c3t3 an instance of `C3T3`
+   * @param tm an instance of `TriangleMesh`
+   * @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+   *
+   * \cgalNamedParamsBegin
+   *   \cgalParamNBegin{face_patch_map}
+  *     \cgalParamDescription{a property map with the patch id's associated to the faces of `faces(tm)`}
+  *     \cgalParamType{a class model of `ReadWritePropertyMap` with `boost::graph_traits<TriangleMesh>::%face_descriptor`
+  *                    as key type and the desired property, model of `CopyConstructible` and `LessThanComparable` as value type.}
+  *     \cgalParamDefault{If not provided, faces patch ids are ignored.}
+  *     \cgalParamExtra{The map is updated during the remeshing process while new faces are created.}
```

Comment:
The parameter pack is passed as `const NamedParameters &`. How can the code modify the map, if the pack itself is `const`?

Reply:
The property map is passed by value, and copied, so it works :-)